From 440f1a5ba1dfaf49167152a367ac9577e0cab928 Mon Sep 17 00:00:00 2001
From: Jan Grulich <jgrulich@redhat.com>
Date: Mon, 20 Nov 2023 13:22:49 +0100
Subject: [PATCH 2/4] Fix indentation of buttons when placed on the left side
 (#43)

Fixes #41
---
 src/qadwaitadecorations.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/qadwaitadecorations.cpp b/src/qadwaitadecorations.cpp
index 7a689f3..17dd766 100644
--- a/src/qadwaitadecorations.cpp
+++ b/src/qadwaitadecorations.cpp
@@ -273,6 +273,9 @@ QRectF QAdwaitaDecorations::buttonRect(Button button) const
         xPos += ceButtonWidth * btnPos;
         xPos += ceButtonSpacing * btnPos;
         xPos += margins().left();
+        // We are painting from the left to the right so the real
+        // position doesn't need to by moved by the size of the button.
+        xPos -= ceButtonWidth;
     }
 
     yPos = margins().top();
-- 
2.42.0

