From 0b0106c86b21e0e73b45613798979fb3784681c5 Mon Sep 17 00:00:00 2001
From: Jan Grulich <jgrulich@redhat.com>
Date: Tue, 21 Nov 2023 08:51:48 +0100
Subject: [PATCH 4/4] Add Adwaita icon theme as a fallback

---
 src/qadwaitadecorations.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/qadwaitadecorations.cpp b/src/qadwaitadecorations.cpp
index f53e130..d1ecd2c 100644
--- a/src/qadwaitadecorations.cpp
+++ b/src/qadwaitadecorations.cpp
@@ -177,8 +177,14 @@ void QAdwaitaDecorations::updateColors(bool useDarkColors)

 QString getIconSvg(const QString &iconName)
 {
-    const QStringList themeNames = { QIcon::themeName(), QIcon::fallbackThemeName() };
+    const QStringList themeNames = { QIcon::themeName(), QIcon::fallbackThemeName(),
+                                     QLatin1String("Adwaita") };
+    qCDebug(QAdwaitaDecorationsLog) << "Icon themes: " << themeNames;
+
     for (const QString &themeName : themeNames) {
+        if (themeName.isEmpty())
+            continue;
+
         for (const QString &path : QIcon::themeSearchPaths()) {
             if (path.startsWith(QLatin1Char(':')))
                 continue;
--
2.42.0

