From 9c3037f90b16db0756fe9c8580646a1e48d9c312 Mon Sep 17 00:00:00 2001
From: Jan Grulich <jgrulich@redhat.com>
Date: Sun, 15 Oct 2023 20:20:47 +0200
Subject: Do not use lambda function for DBus response (#35)


diff --git a/src/qadwaitadecorations.cpp b/src/qadwaitadecorations.cpp
index 35daba5..42493fc 100644
--- a/src/qadwaitadecorations.cpp
+++ b/src/qadwaitadecorations.cpp
@@ -111,28 +111,8 @@ void QAdwaitaDecorations::initConfiguration()

     QDBusPendingCall pendingCall = connection.asyncCall(message);
     QDBusPendingCallWatcher *watcher = new QDBusPendingCallWatcher(pendingCall);
-    QObject::connect(
-            watcher, &QDBusPendingCallWatcher::finished, [this](QDBusPendingCallWatcher *watcher) {
-                QDBusPendingReply<QMap<QString, QVariantMap>> reply = *watcher;
-                if (reply.isValid()) {
-                    QMap<QString, QVariantMap> settings = reply.value();
-                    if (!settings.isEmpty()) {
-                        const uint colorScheme =
-                                settings.value(QLatin1String("org.freedesktop.appearance"))
-                                        .value(QLatin1String("color-scheme"))
-                                        .toUInt();
-                        updateColors(colorScheme == 1); // 1 == Prefer Dark
-                        const QString buttonLayout =
-                                settings.value(QLatin1String("org.gnome.desktop.wm.preferences"))
-                                        .value(QLatin1String("button-layout"))
-                                        .toString();
-                        if (!buttonLayout.isEmpty()) {
-                            updateTitlebarLayout(buttonLayout);
-                        }
-                    }
-                }
-                watcher->deleteLater();
-            });
+    QObject::connect(watcher, &QDBusPendingCallWatcher::finished, this,
+                     &QAdwaitaDecorations::onSettingsReceived);

     QDBusConnection::sessionBus().connect(
             QString(), QLatin1String("/org/freedesktop/portal/desktop"),
@@ -217,6 +197,28 @@ void QAdwaitaDecorations::updateTitlebarLayout(const QString &layout)
     forceRepaint();
 }

+void QAdwaitaDecorations::onSettingsReceived(QDBusPendingCallWatcher *watcher)
+{
+    QDBusPendingReply<QMap<QString, QVariantMap>> reply = *watcher;
+    if (reply.isValid()) {
+        QMap<QString, QVariantMap> settings = reply.value();
+        if (!settings.isEmpty()) {
+            const uint colorScheme = settings.value(QLatin1String("org.freedesktop.appearance"))
+                                             .value(QLatin1String("color-scheme"))
+                                             .toUInt();
+            updateColors(colorScheme == 1); // 1 == Prefer Dark
+            const QString buttonLayout =
+                    settings.value(QLatin1String("org.gnome.desktop.wm.preferences"))
+                            .value(QLatin1String("button-layout"))
+                            .toString();
+            if (!buttonLayout.isEmpty()) {
+                updateTitlebarLayout(buttonLayout);
+            }
+        }
+    }
+    watcher->deleteLater();
+}
+
 void QAdwaitaDecorations::settingChanged(const QString &group, const QString &key,
                                          const QDBusVariant &value)
 {
diff --git a/src/qadwaitadecorations.h b/src/qadwaitadecorations.h
index 6be57ac..2fd162f 100644
--- a/src/qadwaitadecorations.h
+++ b/src/qadwaitadecorations.h
@@ -30,6 +30,7 @@
 using namespace QtWaylandClient;

 class QDBusVariant;
+class QDBusPendingCallWatcher;
 class QPainter;

 class QAdwaitaDecorations : public QWaylandAbstractDecoration
@@ -74,6 +75,7 @@ protected:
 #endif

 private Q_SLOTS:
+    void onSettingsReceived(QDBusPendingCallWatcher *watcher);
     void settingChanged(const QString &group, const QString &key, const QDBusVariant &value);

 private:
